src/service/tcp.rs

```diff
@@ -3,7 +3,7 @@
 
 use std::{
     fmt,
-    io::{Error, ErrorKind},
+    io::{self, Error, ErrorKind},
```

Comment:
Only importing `io` and using `io::Error` and `io::ErrorKind` explicitly would be more consistent and less confusing.